tests/general/views/test_consolidated_profile_view.py

```diff
@@ -0,0 +1,566 @@
+            self.multiple_datasetdata(dataset, self.geo2, data["geo2"])
+            self.multiple_datasetdata(dataset, geo3, data["geo3"])
+            indicator = self.create_indicator(dataset, "gender")
+            self.create_key_metrics(
+                self.profile, indicator, "female", "metric", "sibling"
+            )
+
+        response = self.get(
+            'all-details',
+            profile_id=self.profile.pk,
+            geography_code=self.geo1.code,
+            data={
+                'format': 'json',
+            },
+        )
+        self.assert_http_200_ok()
+        metrics = response.data["profile"]["profile_data"]["category-female"]["subcategories"]["subcategory-female"]["key_metrics"]
+        assert metrics == [
+            {'label': 'metric', 'value': 0.3333333333333333, 'method': 'sibling'}
+        ]
+
+        # Request with version2
+        response = self.get(
+            'all-details',
+            profile_id=self.profile.pk,
+            geography_code=self.geo1.code,
+            data={
+                'version': version2.name,
+                'format': 'json',
+            },
+        )
+        self.assert_http_200_ok()
+        metrics = response.data["profile"]["profile_data"]["category-female"]["subcategories"]["subcategory-female"]["key_metrics"]
+        assert metrics == [
+            {'label': 'metric', 'value': 0.25, 'method': 'sibling'}
+        ]
+
+
+class TestParentLayersData(ConsolidatedProfileViewBase):
+
+    def test_version_parent_and_siblings(self):
+        version1, geography, hierarchy, profile = self.create_base_data(
+            "v2", None, self.hierarchy, self.profile
+        )
+
+        data = [
+            {"gender": "male", "age": "15", "count": 1},
+            {"gender": "female", "age": "16", "count": 2},
+        ]
+        indicator = self.create_versioned_data(
+            version1, geography, profile, data=data
+        )
+
+        response = self.get(
+            'all-details',
+            profile_id=profile.pk,
+            geography_code=geography.code,
+            data={
+                'version': 'v2',
+                'format': 'json'
+            },
+        )
+        self.assert_http_200_ok()
+
+        # assert parents
+        geo_data = response.data["profile"]["geography"]
+        assert len(geo_data["parents"]) == 2
+        assert geo_data["parents"][0]["code"] == self.root.code
+        assert geo_data["parents"][1]["code"] == self.geo1.code
+
+        # Assert parent & siblings and parent siblings
+        parent_layers = response.data["parent_layers"]
+        assert len(parent_layers) == 2
+        assert len(parent_layers[0]["features"]) == 1
+        assert parent_layers[0]["features"][0]["properties"]["code"] == self.geo1.code
+        assert len(parent_layers[1]["features"]) == 1
+        assert parent_layers[1]["features"][0]["properties"]["code"] == geography.code
```

Comment:
again, the hierarchy of geos in each version is critical to quickly be able to understand this test, and to be able to edit it correctly in future.